packages/core-modules/contracts/modules/CoreElectionModule.sol

```diff
@@ -152,24 +157,30 @@ contract CoreElectionModule is IElectionModule, ElectionStorage, OwnableMixin {
 
         uint votePower = ERC20(store.electionTokenAddress).balanceOf(msg.sender);
 
-        for (uint i = 0; i < candidates.length; i++) {
-            address candidate = candidates[i];
+        address[] memory candidates = new address[](sortedCandidates.length);
 
-            if (electionData.nomineeIndexes[candidate] == 0) {
-                revert InvalidCandidate(candidate);
+        for (uint i = 0; i < sortedCandidates.length; i++) {
+            address candidate = sortedCandidates[i];
+            uint position = positions[i];
```

Comment:
Check if position > candidates.length?

Reply:
Done, we are not validating that the candidates length is the same as priorities length: https://github.com/Synthetixio/synthetix-v3/pull/572/files#diff-69650d386e562b4dac992709531657fe63fedd176cde00f236b7b6271333c70eR160-R162

Reply:
You mean you _are_ comparing the lengths?

In any case, how does this solve the case I'm referring to? What if the arrays are 5 in length and the priorities array contains the number 99999?

Reply:
| You mean you are comparing the lengths?
Yup
| In any case, how does this solve the case I'm referring to? What if the arrays are 5 in length and the priorities array contains the number 99999?
And yes, sorry I miss matched the validations, but the priorities number validation is here: https://github.com/Synthetixio/synthetix-v3/pull/572/files#diff-69650d386e562b4dac992709531657fe63fedd176cde00f236b7b6271333c70eR193-R195
Which checks for:
```solidity
            if (priority > numericallySortedCandidates.length - 1) {
                revert InvalidCandidatePriority(priority);
            }
```